### PR TITLE
Add row limit to lookup table uploads

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

https://dimagi.atlassian.net/browse/SAAS-16223

As discussed in the ticket, there aren't any existing lookup tables with more than 500k rows, so that seems like a very safe limit to start with. The risky part of this PR is that it adds an explicit limit to the WorkbookJSONReader class, currently set to 1 million. This limit doesn't have to be set to anything necessarily, but I figured in the spirit of implementing limits, putting a limit like this in code seems ideal. On the flip side, it could be viewed as a lazy way of adding a limit to any code that uses this class under the hood (like scheduling). 

### QA Plan

No

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/util/workbook_json/excel.py [hunk 1]
```diff
@@ -226,10 +217,19 @@ def _convert_float(value):
...
         super(WorksheetJSONReader, self).__init__(iterator())
 
+    def row_count(self):
+        def parse_dimension(dimension):
+            import re
+            match = re.search(r'(\d+)', dimension)
+            if match:
+                return int(match.group(1))
+            return 0
+        return parse_dimension(self.worksheet.calculate_dimension())
```
Comment (millerdev) [inline corehq/util/workbook_json/excel.py:227]: Why use an inner function here? What do you think of this instead?
```suggestion
        import re
        dimension = self.worksheet.calculate_dimension()
        match = re.search(r'(\d+)', dimension)
        return int(match.group(1)) if match else 0
```

Also, what does the result of `self.worksheet.calculate_dimension()` look like? Does it contain other text in addition to the number?

**Review by millerdev — APPROVED** [review]

File: corehq/util/workbook_json/excel.py (hunk 1 above)
Comment (gherceg) [reply corehq/util/workbook_json/excel.py:227]: Yeah I definitely meant to add tests around this function and agree the inner function isn't necessary. Thanks for the feedback.

**Comment by gherceg** [comment]
I think I actually want to replace the implementation of `_max_row` [here](https://github.com/dimagi/commcare-hq/blob/3a06254a9e988e20ebf075a98d5e3f49c8e6fe1f/corehq/util/workbook_reading/adapters/xlsx.py#L59-L76) to use dimensions instead.

**Comment by gherceg** [comment]
Just kidding. After reading some source code I realized the dimensions weren't going to work the way I had hoped, though there is some room for a deeper optimization. My understanding is that dimensions are unreliably set depending on the application that created the file in the first place. So we could rework our code to check for the total size of the workbook/worksheet when first loading it, using the dimensions if they are set, otherwise manually iterating over rows, whereas we currently just iterate over all rows to determine table sizes.

File: corehq/util/workbook_json/excel.py (hunk 1 above)
Comment (gherceg) [reply corehq/util/workbook_json/excel.py]: Ended up removing this entirely. See the conversation and commits below.

**Review by millerdev — DISMISSED** [review]

**Review by millerdev — APPROVED** [review]
Just kidding. :smile: Apparently this is complete.

**Pull request merged**